packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -134,17 +135,23 @@ export const useEnsembleAction = (
     options as UseExecuteCodeActionOptions,
   );
   const navigateScreen = useNavigateScreen(action.navigateScreen, options);
+  const showToast = useShowToast(action.showToast);
   const navigateModalScreen = useNavigateModalScreen(
     action.navigateModalScreen,
     options as null,
   );
-  const showToast = useShowToast(action.showToast);
+  const closeAllDialogs = useCloseAllDialogs();
+
   return (
-    invokeApi ||
-    executeCode ||
-    navigateScreen ||
-    navigateModalScreen ||
-    showToast
+    (action.invokeApi && invokeApi) ||
+    (action.executeCode && executeCode) ||
+    (action.navigateScreen && navigateScreen) ||
+    (action.showToast && showToast) ||
+    (action.navigateModalScreen && navigateModalScreen) ||
+    (keys(action).find((key) => key === "closeAllDialogs") &&
```

Comment:
Why do you need to match keys here?

Reply:
it was giving a weird behaviour for me. it was calling useNavigateModalScreen even though I requested showToast or closeAllDialogs, so I went with this approach :)